@vates/nbd-client/index.js

```diff
@@ -266,20 +309,29 @@ module.exports = class NbdClient {
       }
     }
     const readAhead = []
-    const readAheadMaxLength = 10
+    const readAheadMaxLength = this.#readAhead
+    const makeReadBlockPromise = (index, size) => {
+      const promise = pRetry(() => this.readBlock(index, size), {
+        tries: this.#readBlockRetries,
+        onRetry: async err => {
+          console.warn('will retry reading block ', index, err)
+          await this.reconnect()
```

Comment:
Do we have to disconnect/connect on failure?

Can't we simply attempts to connect if a disconnection has been detected?

Reply:
If I fail to read data from the host, I am not really sure what will be the next content pushed on the socket. I think it's better to restart from a clean situation, even if it means emptying any successfull blocks in the read ahead cache